### ISSUE TITLE
.Quit() is not working when an exception happened

### ISSUE THREAD

**Issue by MonsterMMORPG** [SeleniumHQ/selenium#4054]
I am using windows 8.1
Other components 
- Selenium.Support 3.4.0
- Selenium.WebDriver 3.4.0
- Selenium.WebDriver.ChromeDriver 2.29.0
- c# .net 4.6.2 
- Microsoft Visual Studio Community 2017 RC

The quit method is not working if an exception already happened with chromium driver

First this exception is captured when trying to navigate an URL

URL navigation : 

```
            ChromeDriver cdDrive = new ChromeDriver();
            cdDrive.Navigate().GoToUrl("https://translate.google.com/#en/ar/scuba");
            cdDrive.Navigate().Refresh();
```

```
A exception with a null response was thrown sending an HTTP request to the remote WebDriver server for URL http://localhost:21959/session/e58d72de4345d7653b5178447c8a965f/refresh. The status of the exception was ReceiveFailure, and the message was: The underlying connection was closed: An unexpected error occurred on a receive.
   at OpenQA.Selenium.Remote.HttpCommandExecutor.CreateResponse(WebRequest request)
   at OpenQA.Selenium.Remote.HttpCommandExecutor.Execute(Command commandToExecute)
   at OpenQA.Selenium.Remote.DriverServiceCommandExecutor.Execute(Command commandToExecute)
   at OpenQA.Selenium.Remote.RemoteWebDriver.Execute(String driverCommandToExecute, Dictionary`2 parameters)
   at OpenQA.Selenium.Remote.RemoteNavigator.Refresh()
   at database_builder.csGoogleProcessor.processGoogleTranslate(Int64 irProcessedWordId, String srWord, enLanguageCodes enSource, enLanguageCodes enTarget, Int32 irWordDepth, String& srFetchUrl, String& srSourceFileName) in D:\77 sozluk projesi\database_builder\csGoogleProcessor.cs:line 222
System.Net.WebException: The underlying connection was closed: An unexpected error occurred on a receive. ---> System.IO.IOException: Unable to read data from the transport connection: An existing connection was forcibly closed by the remote host. ---> System.Net.Sockets.SocketException: An existing connection was forcibly closed by the remote host
   at System.Net.Sockets.Socket.Receive(Byte[] buffer, Int32 offset, Int32 size, SocketFlags socketFlags)
   at System.Net.Sockets.NetworkStream.Read(Byte[] buffer, Int32 offset, Int32 size)
   --- End of inner exception stack trace ---
   at System.Net.Sockets.NetworkStream.Read(Byte[] buffer, Int32 offset, Int32 size)
   at System.Net.PooledStream.Read(Byte[] buffer, Int32 offset, Int32 size)
   at System.Net.Connection.SyncRead(HttpWebRequest request, Boolean userRetrievedStream, Boolean probeRead)
   --- End of inner exception stack trace ---
   at System.Net.HttpWebRequest.GetResponse()
   at OpenQA.Selenium.Remote.HttpCommandExecutor.CreateResponse(WebRequest request)
   at System.Net.HttpWebRequest.GetResponse()
   at OpenQA.Selenium.Remote.HttpCommandExecutor.CreateResponse(WebRequest request)
```

When that above exception captured, i want to close existing instance and re-start. Because it doesn't work anymore

This below Quit method is called

`       cdDrive.Quit();`

However Quit method fails miserably with below exception
```

A exception with a null response was thrown sending an HTTP request to the remote WebDriver server for URL http://localhost:21959/session//window. The status of the exception was ConnectFailure, and the message was: Unable to connect to the remote server
   at OpenQA.Selenium.Remote.HttpCommandExecutor.CreateResponse(WebRequest request)
   at OpenQA.Selenium.Remote.HttpCommandExecutor.Execute(Command commandToExecute)
   at OpenQA.Selenium.Remote.DriverServiceCommandExecutor.Execute(Command commandToExecute)
   at OpenQA.Selenium.Remote.RemoteWebDriver.Execute(String driverCommandToExecute, Dictionary`2 parameters)
   at database_builder.csGoogleProcessor.processGoogleTranslate(Int64 irProcessedWordId, String srWord, enLanguageCodes enSource, enLanguageCodes enTarget, Int32 irWordDepth, String& srFetchUrl, String& srSourceFileName) in D:\77 sozluk projesi\database_builder\csGoogleProcessor.cs:line 239
System.Net.WebException: Unable to connect to the remote server ---> System.Net.Sockets.SocketException: No connection could be made because the target machine actively refused it 127.0.0.1:21959
   at System.Net.Sockets.Socket.DoConnect(EndPoint endPointSnapshot, SocketAddress socketAddress)
   at System.Net.ServicePoint.ConnectSocketInternal(Boolean connectFailure, Socket s4, Socket s6, Socket& socket, IPAddress& address, ConnectSocketState state, IAsyncResult asyncResult, Exception& exception)
   --- End of inner exception stack trace ---
   at System.Net.HttpWebRequest.GetResponse()
   at OpenQA.Selenium.Remote.HttpCommandExecutor.CreateResponse(WebRequest request)
   at System.Net.HttpWebRequest.GetResponse()
   at OpenQA.Selenium.Remote.HttpCommandExecutor.CreateResponse(WebRequest request)
```

In short, i have to be able to close failed instances so i can start new instnances and continue to work. But this way, the failed instances never get closed, thus, they keep consuming memory in my system. Need your help. 

questions to solve

1 : when i start new instance, any way to get its process IDs so i can kill directly by process ID?
- e.g. after this method called  ChromeDriver cdDrive = new ChromeDriver(); can i gets that instances process IDs both with chromdriver.exe and opened google chrome exe?

2 : any way to find out exception happened instance and kill with process ID again

Thank you for answers

**Comment by MonsterMMORPG** [comment]
@jimevans @lmtierney @thred @zniff @brunnerh

**Comment by AutomatedTester** [comment]
Please log chromedriver issues with the chromedriver project.

https://sites.google.com/a/chromium.org/chromedriver/help

Please be sure to include a completely reproducible test script for them, without a way to reproduce the issue you are seeing there is no good way for them to fix the problem. A reproducible test script includes a WebDriver script and a link to the page in question. If you can not provide a link, create one using [jsfiddle.net](https://jsfiddle.net)

**Comment by MonsterMMORPG** [comment]
@AutomatedTester so you say i would not get such errors if i were using firefox?

**Comment by AutomatedTester** [comment]
@MonsterMMORPG If you can reproduce feel free to raise a GeckoDriver bug. Please note though for issues please provide a [concise reproducible test case](http://sscce.org/) and describe what results you are seeing and what results you expect.

When raising a bug, please follow the details in [CONTRIBUTING.md](https://github.com/mozilla/geckodriver/blob/master/CONTRIBUTING.md#issue-reports)

**Comment by MonsterMMORPG** [comment]
@AutomatedTester i cant reproduce because it happens during the actual course of the application. But i am providing full error log. What else can be needed?

**Comment by AutomatedTester** [comment]
@MonsterMMORPG your error log is great for raising a ChromeDriver issue.
